aci/login/helper.go

```diff
@@ -39,27 +39,27 @@ var (
 )
 
 type apiHelper interface {
-	queryToken(data url.Values, tenantID string) (azureToken, error)
-	openAzureLoginPage(redirectURL string) error
+	queryToken(ce CloudEnvironment, data url.Values, tenantID string) (azureToken, error)
+	openAzureLoginPage(redirectURL string, ce CloudEnvironment) error
 	queryAPIWithHeader(ctx context.Context, authorizationURL string, authorizationHeader string) ([]byte, int, error)
-	getDeviceCodeFlowToken() (adal.Token, error)
+	getDeviceCodeFlowToken(ce CloudEnvironment) (adal.Token, error)
```

Comment:
I made the change, it is a bit better, no more `CloudEnvironment` passing to helper functions. However, in the login service code, we cannot initialize `apiHelper` right away because the cloud environment information is not available until we read the token from the store, typically. So in the login service code the `apiHelper` is created as necessary and passed around.

Reply:
OK, I thought this would be easier, but I understand the constraints